worker/machineactions/worker.go

```diff
@@ -103,30 +109,59 @@ func (h *handler) Handle(_ <-chan struct{}, actionsSlice []string) error {
 		if err != nil {
 			return errors.Annotatef(err, "could not retrieve action %s", actionId)
 		}
-
-		err = h.config.Facade.ActionBegin(actionTag)
-		if err != nil {
-			return errors.Annotatef(err, "could not begin action %s", action.Name())
-		}
-
-		// We try to handle the action. The result returned from handling the action is
-		// sent through using ActionFinish. We only stop the loop if ActionFinish fails.
-		var finishErr error
-		results, err := h.config.HandleAction(action.Name(), action.Params(), action.Parallel(), action.ExecutionGroup())
-		if err != nil {
-			finishErr = h.config.Facade.ActionFinish(actionTag, params.ActionFailed, nil, err.Error())
-		} else {
-			finishErr = h.config.Facade.ActionFinish(actionTag, params.ActionCompleted, results, "")
-		}
-		if finishErr != nil {
-			return errors.Trace(finishErr)
-		}
+		// TODO(wallyworld) - use a thread pool to put a cap on the max number of parallel actions.
```

Comment:
You shouldn't need a thread pool for this - this is pretty simple to do with a couple of extra lines of code using a buffered "limiter" channel with a size of the max concurrency you want. See: https://play.golang.org/p/PMW0rQv0wjO

Reply:
Good suggestion, done